syncserver/src/tokenserver/handlers.rs

```diff
@@ -121,7 +121,7 @@ struct UserUpdates {
 
 async fn update_user(
     req: &TokenserverRequest,
-    db: Box<dyn Db>,
+    db: Box<dyn DbTrait>,
```

Comment:
It would be nice to get rid of these trait objects now that we don't need to heterogeneously handle MySQL and Spanner database pools, but we still need them to handle `MockDb` and `MockDbPool` in the unit tests, since some unit tests use the mock data structures and some use the real database data structures

Reply:
Going to say that we probably shouldn't. I know YAGNI is a thing, but if Autopush is any guide, having a level of database abstraction can be a benefit in the future. Even if it's just encapsulating one "type" right now, there's a reasonable chance it may wrap future ones later. (e.g. we have to move to a new provider, or we need to move to a different data store, etc.)